infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -259,54 +156,73 @@ public synchronized void handleOrchestratorMsg(@Nonnull CorfuPayloadMsg<Orchestr
     @ServerHandler(type = CorfuMsgType.MANAGEMENT_BOOTSTRAP_REQUEST)
     public synchronized void handleManagementBootstrap(CorfuPayloadMsg<Layout> msg,
                                                        ChannelHandlerContext ctx, IServerRouter r) {
-        if (latestLayout != null) {
+        if (serverContext.getManagementLayout() != null) {
             // We are already bootstrapped, bootstrap again is not allowed.
             log.warn("Got a request to bootstrap a server which is already bootstrapped, "
                     + "rejecting!");
             r.sendResponse(ctx, msg, new CorfuMsg(CorfuMsgType.MANAGEMENT_ALREADY_BOOTSTRAP_ERROR));
         } else {
             log.info("Received Bootstrap Layout : {}", msg.getPayload());
-            safeUpdateLayout(msg.getPayload());
+            serverContext.saveManagementLayout(msg.getPayload());
             r.sendResponse(ctx, msg, new CorfuMsg(CorfuMsgType.ACK));
+
         }
     }
 
     /**
-     * Trigger to start the failure handler.
+     * Triggers the failure handler.
+     * The msg contains the failed/defected nodes.
      *
-     * @param msg corfu message containing MANAGEMENT_START_FAILURE_HANDLER
+     * @param msg corfu message containing MANAGEMENT_FAILURE_DETECTED
      * @param ctx netty ChannelHandlerContext
      * @param r   server router
      */
-    @ServerHandler(type = CorfuMsgType.MANAGEMENT_START_FAILURE_HANDLER)
-    public synchronized void initiateFailureHandler(CorfuMsg msg, ChannelHandlerContext ctx,
-                                                    IServerRouter r) {
+    @ServerHandler(type = CorfuMsgType.MANAGEMENT_FAILURE_DETECTED)
+    public synchronized void handleFailureDetectedMsg(CorfuPayloadMsg<DetectorMsg> msg,
+                                                      ChannelHandlerContext ctx, IServerRouter r) {
 
         // This server has not been bootstrapped yet, ignore all requests.
         if (!checkBootstrap(msg, ctx, r)) {
             r.sendResponse(ctx, msg, new CorfuMsg(CorfuMsgType.MANAGEMENT_NOBOOTSTRAP_ERROR));
             return;
         }
 
-        if (!startFailureHandler) {
-            startFailureHandler = true;
-            log.info("Initiated Failure Handler.");
+        log.info("handleFailureDetectedMsg: Received DetectorMsg : {}", msg.getPayload());
+
+        DetectorMsg detectorMsg = msg.getPayload();
```

Comment:
Update comments for this new logic. It is better.

Reply:
Done.